src/sqlfluff/dialects/dialect_clickhouse.py

```diff
@@ -71,6 +213,7 @@ class FromExpressionElementSegment(ansi.FromExpressionElementSegment):
                 Ref("SamplingExpressionSegment"),
                 Ref("JoinLikeClauseGrammar"),
                 Ref.keyword("Final"),
```

Comment:
Not your issue but just noticed this. IT doesn't need the Ref.keyword and `Final` should be in uppercase.

```suggestion
                FINAL,
```